### DISCUSSION TITLE
Any way to use the HTTP provider for Infura connections?

### DISCUSSION THREAD

**Discussion by milaabl** [WalletConnect/walletconnect-monorepo#4195] [Developer Support] [answered]
I'm troubleshooting an Infura-specific problem, and as Infura's support suggested, I should try connecting to an HTTP endpoint. Currently a WebSocket connection is automatically established with the following config:

```javascript
const provider = await EthereumProvider.init({
                    projectId: process.env.NEXT_PUBLIC_INFURA_ID!,
                    showQrModal: true,
                    optionalChains: [1, 5, 11155111],
                });
await provider.enable();
```

Is there any option I can pass to the EthereumProvider in order to connect via HTTP instead?

Resources:

- https://docs.infura.io/networks/ethereum/how-to/make-requests

**Comment by glitch-txs** [comment, marked as answer]
Hi Mila! You can use the `rpcMap` option in the `init` function :)


```ts
rpcMap: {
 1: "yourRpcUrl...",
}
```

The key must be the chain ID you want to target and then value the RPC URL you want to assign to it

**Reply by milaabl** [reply]
Ahh, got it, thank you!! Will try!
